### PR TITLE
Fix incorrect file name regex logic

### PR DESCRIPTION
**Pull request by JSpiner**
When I run [nexus](https://www.sonatype.com/products/repository-oss) with `nexus run`, I got the following error.

> Caused by: java.io.IOException: Directory '/Users/smith-debug.jspiner/nexus-3.34.1-01-mac/sonatype-work/nexus3/tmp/nexus-extdirect' could not be created
	at org.apache.commons.io.FileUtils.openOutputStream(FileUtils.java:2436)
	at org.apache.commons.io.FileUtils.writeStringToFile(FileUtils.java:3356)
	at org.apache.commons.io.FileUtils.writeStringToFile(FileUtils.java:3305)

My user name is `smith.jspiner` and file path is `/Users/smith.jspiner/nexus-.....`. 
But strangely, nexus was trying with `/Users/smith-debug.jspiner/nexus-....`. 😭 

When I checked the source code, there was an error in changing the file name for debug.

I modified the regex logic and added the test code. 
Please review it. 🙏

### CODE REVIEW

**Comment by JSpiner** [comment]
@dbradicich @ataylor284 @thejosh00 
Hi! Can I ask for a review? 🙏 
I am unable to use nexus because of this issue. 😭

File: directjngine/src/main/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGenerator.java
Comment (ataylor284) [inline directjngine/src/main/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGenerator.java:171]: `file.replaceFirst(".js$", "-debug.js");` might be a better way to express this.

File: directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java
Comment (ataylor284) [inline directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java:11]: I'd just make `getDebugFileName` package protected (no modifier) instead of using reflection to invoke for testing.

**Review by ataylor284 — CHANGES_REQUESTED** [review]

File: directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java
Comment (dbradicich) [inline directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java:1]: I believe we need to retain the same license header as used for other source files, i.e. https://github.com/sonatype/directjngine/blob/master/directjngine/src/test/java/com/softwarementors/extjs/djn/gson/DefaultGsonBuilderConfiguratorTest.java

**Review by dbradicich — COMMENTED** [review]
@JSpiner Thanks for pushing up a UT as well :) will surely get merged after the comments resolved, we are going to be doing a new release to use in nxrm 3.37 soon and would surely like to include this for you

File: directjngine/src/main/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGenerator.java
Comment (JSpiner) [reply directjngine/src/main/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGenerator.java]: Much better! 👍 
Applied in 9eb5223

File: directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java
Comment (JSpiner) [reply directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java]: I was confused because I didn't know the test rules of the project, but it's good that I can use protected function. 👍 
Applied in 68d3969

File: directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java
Comment (JSpiner) [reply directjngine/src/test/java/com/softwarementors/extjs/djn/jscodegen/CodeFileGeneratorTest.java:25]: Oops I missed it. 😅 
Applied in d7a1d5c

**Comment by nblair** [comment]
Hi @JSpiner - looks like the code is ready for us to merge. Before we can do that, we need to you sign our Contributor License Agreement, found here:

https://sonatypecla.herokuapp.com/sign-cla

Once you're signature is in we can merge and include in the next release - thank you!

**Review by ataylor284 — APPROVED** [review]

**Review by dbradicich — APPROVED** [review]
Looks good to me, I almost forgot the CLA needs to signed prior to accepting the PR, but codewise looks good to me :) 

Please see this link for the CLA, just comment back here when done and we should be able to get merged for next release (earliest release would be 3.37 of nxrm fyi)

**Comment by JSpiner** [comment]
Hi! Can it be merged?
I can't use Nexus because of this issue. 😭

**Comment by Blacktiger** [comment]
Merged, but it won't get included in Nexus Repository until the next release and I'm not sure when that will be but we missed getting this into 3.37 unfortunately. You could also build this locally and replace the jar in your nexus instance while you wait for a new release.

**Pull request merged**
